### DISCUSSION TITLE
Attempting to use swc for only certain parts of build (example: preset-env)

### DISCUSSION THREAD

**Discussion by imjoshin** [swc-project/swc#4859] [Q&A]
Hello!

I'm currently working in a build environment that is heavily reliant on babel. I'm hoping to try to swap certain pieces of the build out with swc without needing to convert everything, the first target being `babel-preset-env`. I see SWC has support for the preset-env functionality, but I'm curious if it's possible to target that specific part of the swc process through `swc-loader`. At which point, we would hand the rest off to subsequent loaders.

I was looking through these docs and did not find exactly what I'm looking for:
- https://swc.rs/docs/configuration/compilation
- https://swc.rs/docs/usage/core#options

Here's an example WIP diff: https://github.com/gatsbyjs/gatsby/compare/preset-env-with-swc

**Comment by kwonoj** [comment]
afaik I don't think we have graceful way to make things work magically for this. Probably best bet is puzzling with include / exclude in preset-env (https://github.com/swc-project/swc/blob/main/node-swc/src/types.ts#L598-L602=) to swc do best attempt to skip certain transforms.

**Reply by kwonoj** [reply]
I'm curious if this would works for you, and would like to hear if it's not. Gradual migration path is somewhat interesting topic and potentially helpful if we could find generic way.

**Reply by imjoshin** [reply]
Thanks for the response! I've given a shot with no success. Unfortunately given the ordering (currently) needed in the build process, we require the `preset-env` transform to be done early on, which triggers swc's other transforms as we haven't gotten to those yet. Not to say that we can't refactor or rethink the way our flow is working currently, but just that we don't have a way forward right now.

**Reply by kwonoj** [reply]
It'd be helful if you could share some more details how does transpilation passes are setup currently & what's the current goal to achieve. I can't promise it can be solved easily or not though 😅

**Comment by Austaras** [comment]
Maybe it's possible to chain a swc after babel transforms jsx and other things, but I'm not sure about potential performance.
